markdown-mode.el

```diff
@@ -8784,6 +8785,11 @@ at the END of code blocks."
                       (lambda (_parent-buffer _beg _end)
                         (funcall mode)))
                      (indirect-buf (edit-indirect-region begin end 'display-buffer)))
+                (when (member mode '(shell-script-mode sh-mode))
+                  (with-current-buffer indirect-buf
+                    (if (member lang '("screen" "shell"))
+                        (sh-set-shell "sh")
+                      (sh-set-shell lang)))) ;; reset `sh-shell' in indirect buffer
```

Comment:
Should `lang` be checked as valid shell ? Some code blocks uses `shell-script`. This is not valid shell name. And some shell like `zsh` or `fish` does not recognize on Windows.

https://github.com/github/linguist/blob/master/lib/linguist/languages.yml

Reply:
I use `zsh` as $SHELL on MacOS, however , when I edit `sh` or `bash` code block, the indirect-buffer always identified as `zsh`. Of course , I could set `sh-shell-file` to `bash` as default , but when I edit `zsh` code block, It will identified as `bash` .